lib/src/iterable.dart

```diff
@@ -1005,3 +1005,18 @@ extension IterableX<E> on Iterable<E> {
     return [t, f];
   }
 }
+
+extension IterableFutureX<E> on Iterable<Future<E>> {
+  /// Create a stream from a group of futures.
+  ///
+  /// The stream reports the results of the futures on the stream in the order
+  /// in which the futures complete.
+  /// Each future provides either a data event or an error event,
+  /// depending on how the future completes.
+  ///
+  /// If some futures have already completed when `Stream.fromFutures` is called,
+  /// their results will be emitted in some unspecified order.
+  ///
+  /// When all futures have completed, the stream is closed.
+  Stream<E> asStreamAwaited() => Stream.fromFutures(this);
```

Comment:
I'm not 100% happy with the naming. `as*` methods for me do things like casting. But `fromFeatures` does a bit more. I'd like to have a verb. What do you think about `streamed()`?

Reply:
Actually your concern doesn't make much sense. Remember that we have both `asMap` and `toList` method (one for List and the other for Iterables). 
In Dart we usually call `as` or `to` when we want to transform things into other things (like the examples above).